Tella/Scenes/Uwazi/Views/Entity/RenderPropertyComponentView.swift

```diff
@@ -9,61 +9,57 @@
-                            showMandatory: $prompt.showMandatoryError,
-                            shouldRender:shouldRenderPrompt(forType: prompt.type),
-                            showClear: prompt.showClear ?? false,
-                            onClearAction: { entityViewModel.clearValues(forId: prompt.id ?? "")}
-        ) {
-            renderPropertyComponent(
-                prompt: prompt
-            )
-            if(prompt.showMandatoryError) {
-                UwaziEntityMandatoryTextView()
-            }
-        }
-    }
-    
-    @ViewBuilder
-    private func renderPropertyComponent(prompt: UwaziEntryPrompt) -> some View {
-        switch UwaziEntityPropertyType(rawValue: prompt.type) {
-        case .dataTypeText, .dataTypeNumeric, .dataTypeMarkdown:
-            UwaziTextWidget(value: prompt.value)
-        case .dataTypeSelect, .dataTypeMultiSelect:
-            UwaziSelectWidget(value: prompt.value)
-                .environmentObject(prompt)
-                .environmentObject(entityViewModel)
-        case .dataTypeMultiFiles:
-            SupportingFileWidget()
-                .environmentObject(prompt)
-                .environmentObject(sheetManager)
-                .environmentObject(entityViewModel)
-        case .dataTypeMultiPDFFiles:
-            PrimaryDocuments()
-                .environmentObject(prompt)
-                .environmentObject(sheetManager)
-                .environmentObject(entityViewModel)
-        case .dataTypeDivider:
-            UwaziDividerWidget()
-        case .dataTypeDate:
-            UwaziDatePicker(value: prompt.value)
-                .environmentObject(prompt)
-        default:
-            EmptyView()
+        
+        if shouldRenderPrompt(forType: prompt.type.rawValue) {
+            VStack() {
+                
+                UwaziEntityTitleView(title: prompt.question,
+                                     isRequired: prompt.required ?? false,
+                                     showClear: prompt.showClear,
+                                     onClearAction: {
+                    prompt.empty()
+                    entityViewModel.publishUpdates()
+                })
+                
+                switch prompt.type {
+                case .dataTypeText, .dataTypeNumeric, .dataTypeMarkdown:
+                    UwaziTextWidget(prompt: prompt as! UwaziTextEntryPrompt)
+                case .dataTypeSelect, .dataTypeMultiSelect:
+                    UwaziSelectWidget(prompt: prompt as! UwaziSelectEntryPrompt)
+                case .dataTypeMultiFiles:
+                    SupportingFileWidget(prompt: prompt as! UwaziFilesEntryPrompt)
+                case .dataTypeMultiPDFFiles:
+                    PrimaryDocuments(prompt: prompt as! UwaziFilesEntryPrompt)
+                case .dataTypeDivider:
+                    UwaziDividerWidget()
+                case .dataTypeDate:
+                    UwaziDatePicker()
```

Comment:
Currently the date component is breaking because it does not receive the prompt via params.